### PR TITLE
[3rd viz] SQL script for moved-boxes

### PR DESCRIPTION
**Pull request by pylipp**

### CODE REVIEW

File: back/boxtribute_server/business_logic/statistics/sql.py [hunk 1]
```diff
@@ -264,25 +264,25 @@
...
             ELSE 0
         END
-    ) AS state_change_items
+    ) AS items_count
 FROM FinalResult t
 JOIN products p ON p.id = t.product
 JOIN locations loc ON loc.id = t.location_id
 LEFT OUTER JOIN tags_relations tr ON tr.object_id = t.box_id AND tr.object_type = "Stock"
+WHERE (t.prev_box_state_id = 1 AND t.box_state_id = 5) OR
+      (t.prev_box_state_id = 5 AND t.box_state_id = 1)
```
Comment (pylipp) [inline back/boxtribute_server/business_logic/statistics/sql.py:257]: try to move it to previous CTE

File: back/boxtribute_server/business_logic/statistics/crud.py [hunk 2]
```diff
@@ -240,24 +245,21 @@ def compute_top_products_checked_out(base_id):
...
     with rank included, grouped by distribution date and product category.
     """
+    _validate_existing_base(base_id)
     selection = Transaction.select(
-        Transaction.created_on.alias("checked_out_on"),
+        fn.DATE(Transaction.created_on).alias("checked_out_on"),
         Transaction.product.alias("product_id"),
         Product.category.alias("category_id"),
         fn.SUM(Transaction.count).alias("items_count"),
+        fn.RANK().over(order_by=[fn.SUM(Transaction.count).desc()]).alias("rank"),
```
Comment (vahidbazzaz) [inline back/boxtribute_server/business_logic/statistics/crud.py:254]: Nice job!

**Review by vahidbazzaz — APPROVED** [review]

**Comment by vahidbazzaz** [comment]
@pylipp Just circling back to our convo – the changes you made totally make sense. I'd be super happy if you could find some time to use UNION so we can get the results for other boxes (with shipment) in a single query. No rush, just whenever you can. Thanks a ton! 😊

**Comment by vahidbazzaz** [comment]
@pylipp Could you also add some comments about our assumptions for box counting? It'll be helpful to document that. Oh, and just a quick note, it'd be awesome if you could include a brief summary of the actual problem statement in the code comments. Thanks!

**Comment by pylipp** [comment]
Good point @vahidbazzaz I've extended the docstring.
A follow-up task is here: https://github.com/boxwise/boxtribute/issues/1203

**Pull request merged**